applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -304,124 +335,87 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
         } else {
-            result = service.listLabels(baseUri.toString(), null, null, null, null);
+            result = service.listKeys(baseUri.toString(), null, null, null, null);
         }
-        return result.flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
-    }
 
-    private Flux<RestPagedResponse<Label>> listLabelsNextSinglePageAsync(@NonNull String nextPageLink) {
-        if (nextPageLink == null) {
-            throw new IllegalArgumentException("Parameter nextPageLink is required and cannot be null.");
-        }
-        String nextUrl = String.format("%s", nextPageLink);
-        return service.listLabelsNext(baseUri.toString(), nextUrl)
-                       .flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+        return result.flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())))
+                .concatMap(this::extractAndFetchKeys);
     }
 
     /**
-     * List all Keys.
+     * Gets all Keys.
      *
-     * @return keys
+     * @param nextPageLink The nextPageLink from the previous successful call to List operation.
+     * @return A stream of Keys from the nextPageLink
      */
-    public Flux<Key> listKeys(KeyLabelFilter filter) {
-        return listKeys(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
-                                                            .flatMapIterable(i -> i));
+    private Flux<Key> listKeys(@NonNull String nextPageLink) {
+        return service.listKeysNext(baseUri.toString(), nextPageLink)
+                .flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())))
+                .concatMap(this::extractAndFetchKeys);
     }
 
-    public <T> Flux<T> listKeys(KeyLabelFilter filter, Function<Flux<RestPagedResponse<Key>>, ? extends Flux<T>> receiver) {
-        Flux<RestPagedResponse<Key>> p = listKeysSinglePageAsync(filter)
-                       .concatMap(page -> {
-                           String nextPageLink = page.nextLink();
-                           if (nextPageLink == null) {
-                               return Flux.just(page);
-                           }
-                           return Flux.just(page).concatWith(listKeysNextAsync(nextPageLink));
-                       });
-        return receiver.apply(p);
+    private Publisher<ConfigurationSetting> extractAndFetchConfigurationSettings(RestPagedResponseImpl<ConfigurationSetting> page) {
+        String nextPageLink = page.nextLink();
+        if (nextPageLink == null) {
+            return Flux.fromIterable(page.items());
+        }
+        return Flux.fromIterable(page.items()).concatWith(listKeyValues(nextPageLink));
```

Comment:
Using a similar form as above, this too could be a useful generic utility method.